File: server/router/report/analysis_report.go
package report

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type AnalysisReportRouter struct{}

// InitAnalysisReportRouter 初始化报表平台·分析报表路由
func (r *AnalysisReportRouter) InitAnalysisReportRouter(Router *gin.RouterGroup) {
	priv := Router.Group("report")
	write := priv.Group("analysisReport").Use(middleware.OperationRecord())
	read := priv.Group("analysisReport")
	{
		write.POST("createAnalysisReport", analysisApi.CreateAnalysisReport)   // 新增
		write.PUT("updateAnalysisReport", analysisApi.UpdateAnalysisReport)    // 更新
		write.DELETE("deleteAnalysisReport", analysisApi.DeleteAnalysisReport) // 删除（级联配置）
		write.POST("copyAnalysisReport", analysisApi.CopyAnalysisReport)       // 复制
		write.POST("saveAnalysisConfig", analysisApi.SaveAnalysisConfig)       // 保存设计器配置
	}
	{
		read.GET("getAnalysisReportList", analysisApi.GetAnalysisReportList)     // 分页
		read.GET("findAnalysisReport", analysisApi.FindAnalysisReport)           // 详情
		read.GET("getAnalysisReportByCode", analysisApi.GetAnalysisReportByCode) // 按编码详情
		read.POST("previewAnalysisReport", analysisApi.PreviewAnalysisReport)    // 预览取数（查询性质，不记操作日志）
	}
}

File: web/src/api/report/analysisReport.js
import service from '@/utils/request'

export const getAnalysisReportList = (params) => service({ url: '/report/analysisReport/getAnalysisReportList', method: 'get', params })
export const findAnalysisReport = (params) => service({ url: '/report/analysisReport/findAnalysisReport', method: 'get', params })
export const getAnalysisReportByCode = (params) => service({ url: '/report/analysisReport/getAnalysisReportByCode', method: 'get', params })
export const createAnalysisReport = (data) => service({ url: '/report/analysisReport/createAnalysisReport', method: 'post', data })
export const updateAnalysisReport = (data) => service({ url: '/report/analysisReport/updateAnalysisReport', method: 'put', data })
export const deleteAnalysisReport = (params) => service({ url: '/report/analysisReport/deleteAnalysisReport', method: 'delete', params })
export const copyAnalysisReport = (data) => service({ url: '/report/analysisReport/copyAnalysisReport', method: 'post', data })
export const saveAnalysisConfig = (data) => service({ url: '/report/analysisReport/saveAnalysisConfig', method: 'post', data })
export const previewAnalysisReport = (data) => service({ url: '/report/analysisReport/previewAnalysisReport', method: 'post', data })

/**
 * @typedef {Object} ValueConfig
 * @property {string} field
 * @property {'SUM'|'COUNT'|'AVG'|'MIN'|'MAX'|'NONE'} aggregation
 * @property {string} [alias]
 * @property {string} [format]   // numeral 风格模板，如 '#,##0.00'
 */
/**
 * @typedef {Object} ReportConfig
 * @property {'pivot'|'table'} sheetType
 * @property {{rows:string[], columns:string[], values:ValueConfig[], valueInRow:boolean}} fields
 * @property {{pagination:{open:boolean,current:number,pageSize:number}, frozenRowHeader:boolean,
 *   showSeriesNumber:boolean, hierarchyType:'grid'|'tree', layoutWidthType:'adaptive'|'colAdaptive'|'compact',
 *   adaptive:boolean, totals:{row:{showGrandTotals:boolean,showSubTotals:boolean,subTotalsDimensions:string[]},
 *   col:{showGrandTotals:boolean,showSubTotals:boolean,subTotalsDimensions:string[]}}, style?:{rowHeight?:number,colWidth?:number}}} options
 * @property {{name:'default'|'dark'|'colorful'|'gray'}} theme
 */

export const AGGREGATION_OPTIONS = [
  { label: '求和 (SUM)', value: 'SUM' },
  { label: '计数 (COUNT)', value: 'COUNT' },
  { label: '平均 (AVG)', value: 'AVG' },
  { label: '最小 (MIN)', value: 'MIN' },
  { label: '最大 (MAX)', value: 'MAX' },
  { label: '不聚合·直显 (NONE)', value: 'NONE' }
]

export const THEME_OPTIONS = [
  { label: '默认', value: 'default' },
  { label: '暗色', value: 'dark' },
  { label: '多彩', value: 'colorful' },
  { label: '灰色', value: 'gray' }
]

export const HIERARCHY_OPTIONS = [
  { label: '平铺', value: 'grid' },
  { label: '树状', value: 'tree' }
]

export const LAYOUT_WIDTH_OPTIONS = [
  { label: '自适应', value: 'adaptive' },
  { label: '列等宽', value: 'colAdaptive' },
  { label: '紧凑', value: 'compact' }
]

export const PAGE_SIZE_OPTIONS = [20, 50, 100]
